File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int last = 0;

        for (int i : section) {
            if (i > last) {
                last = i + m - 1;
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.kt
class Solution {
    fun solution(n: Int, m: Int, section: IntArray): Int {
        var answer: Int = 0
        var last = 0

        for (i in section) {
            if (i > last) {
                last = i + m - 1
                answer += 1
            }
        }

        return answer
    }
}